spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/ErrorHandling.scala

```diff
@@ -16,25 +16,130 @@
 
 package za.co.absa.spark.commons.errorhandling
 
-import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import org.apache.spark.sql.catalyst.expressions.{CaseWhen, Expression}
+import org.apache.spark.sql.functions.when
+import org.apache.spark.sql.{Column, DataFrame}
 import za.co.absa.spark.commons.errorhandling.implementations.submits.{ErrorMessageSubmitOnColumn, ErrorMessageSubmitWithoutColumn}
 import za.co.absa.spark.commons.errorhandling.types._
 
+/**
+ * The basic class of error handling component. Every library that wants to use the component during Spark data
+ * processing should utilize this trait and its methods. The methods serve to record the errors and attach them to the
+ * `DataFrame`. The trait should be an input parameter for such library, perhaps as an implicit.
+ * On the other side the end application provides concrete `ErrorHandling` implementation, that does the actual error
+ * handling by the application desire.
+ * For easy to use ana as examples, a few general implementations are provided in the implementations sub-folder.
```

Comment:
just for the sake of completeness, you can maybe even mention the partials (partial implementations) if you like